sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1176,6 +1198,32 @@ def check_config_error(self, api, channel, states):
 
         return done
 
+    def check_datapath_init_pending(self, api, channel):
+        """
+        Check if the CMIS datapath init is pending
+
+        Args:
+            api:
+                XcvrApi object
+            channel:
+                Integer, a bitmask of the lanes on the host side
+                e.g. 0x5 for lane 0 and lane 2.
+
+        Returns:
+            Boolean, true if any lanes are pending datapath init, otherwise false
+        """
+        pending = False
+        dpinit_pending_dict = api.get_dpinit_pending()
+        for lane in range(self.CMIS_NUM_CHANNELS):
+            if ((1 << lane) & channel) == 0:
+                continue
+            key = "DPInitPending{}".format(lane + 1)
+            if dpinit_pending_dict[key]:
+                pending = True
+                break
+
+        return pending
+
```

Comment:
I think we should check if DpInitPending is set for ALL host lanes and return TRUE. Return False if any of the lanes has DpInitPending=0


<img width="986" alt="image" src="https://user-images.githubusercontent.com/45705344/193944122-fc5b9b3b-edee-4347-af27-6dd2b6bbcbd5.png">